integrationTests/vm/txsFee/freezeAccounts_test.go

```diff
@@ -0,0 +1,384 @@
+	require.Nil(t, err)
+	require.Equal(t, big.NewInt(transferValue), getBalance(testContext, receiverAddress))
+
+	// userAddress can send funds while it just added a guardian
+	returnCode, err := setGuardian(testContext, userAddress, guardianAddress, uuid, delayedSetGuardian)
+	assert.Nil(t, err)
+	assert.Equal(t, vmcommon.Ok, returnCode)
+
+	testActiveGuardian(t, testContext, userAddress, true, nil, nil)
+	testPendingGuardian(t, testContext, userAddress, false, guardianAddress, uuid)
+
+	err = transferFunds(testContext, userAddress, big.NewInt(transferValue), receiverAddress)
+	require.Nil(t, err)
+	require.Equal(t, big.NewInt(transferValue*2), getBalance(testContext, receiverAddress))
+
+	// delay epoch pasts, the pending guardian is now active (but not activated), userAddress can send funds
+	hdr := &block.Header{
+		Epoch: vm.EpochGuardianDelay,
+	}
+	testContext.EpochNotifier.CheckEpoch(hdr)
+
+	testActiveGuardian(t, testContext, userAddress, false, guardianAddress, uuid)
+	testPendingGuardian(t, testContext, userAddress, true, nil, nil)
+
+	err = transferFunds(testContext, userAddress, big.NewInt(transferValue), receiverAddress)
+	require.Nil(t, err)
+	require.Equal(t, big.NewInt(transferValue*3), getBalance(testContext, receiverAddress))
+
+	// userAddress can not send funds while protected without setting the guardian address
+	returnCode, err = guardAccount(testContext, userAddress)
+	assert.Nil(t, err)
+	assert.Equal(t, vmcommon.Ok, returnCode)
+
+	testActiveGuardian(t, testContext, userAddress, false, guardianAddress, uuid)
+	testPendingGuardian(t, testContext, userAddress, true, nil, nil)
+
+	err = transferFunds(testContext, userAddress, big.NewInt(transferValue), receiverAddress)
+	require.ErrorIs(t, err, process.ErrTransactionNotExecutable)
+	require.Contains(t, err.Error(), "not allowed to bypass guardian")
+	require.Equal(t, big.NewInt(transferValue*3), getBalance(testContext, receiverAddress))
+
+	// userAddress can send funds while protected with the guardian address
+	err = transferFundsWithGuardian(testContext, userAddress, big.NewInt(transferValue), receiverAddress, guardianAddress)
+	require.Nil(t, err)
+	require.Equal(t, big.NewInt(transferValue*4), getBalance(testContext, receiverAddress))
+
+	// userAddress can not send funds while protected with a wrong guardian address
+	err = transferFundsWithGuardian(testContext, userAddress, big.NewInt(transferValue), receiverAddress, wrongGuardianAddress)
+	require.ErrorIs(t, err, process.ErrTransactionNotExecutable)
+	require.Contains(t, err.Error(), "mismatch between transaction guardian and configured account guardian")
+	require.Equal(t, big.NewInt(transferValue*4), getBalance(testContext, receiverAddress))
+
+	// userAddress can not send funds while protected with an empty guardian address
+	err = transferFundsWithGuardian(testContext, userAddress, big.NewInt(transferValue), receiverAddress, nil)
```

Comment:
maybe another test to check that sending funds with guardian is not processed while the account is not guarded?

Reply:
done, the test fails :D
Because we have this code:
```GO
func (txProc *baseTxProcessor) verifyGuardian(tx *transaction.Transaction, account state.UserAccountHandler) error {
	if check.IfNil(account) {
		return nil
	}
	if !account.IsGuarded() {
		return nil
	}
	if !txProc.txVersionChecker.IsGuardedTransaction(tx) {
		return txProc.checkGuardedAccountUnguardedTxPermission(tx.GetData(), account)
	}
```
and the code does an early exit on the second if branch. Will discuss how to fix this

Reply:
added fix + unit tests